v3/routes/exercises.rb

```diff
@@ -18,6 +20,14 @@ class Exercises < Core
         status 404 unless problem.exists?
         pg :"exercise/exists", locals: { problem: problem }
       end
+
+      get '/tracks/:id/exercises/:slug' do |id, slug|
+        problem = config.find(id).find(slug)
+        file = Tempfile.new('data.zip')
+        ZipFileGenerator.write(problem.send(:dir), file.path)
+        send_file file.path, type: 'application/octet-stream'
+        file.unlink
```

Comment:
What happens if send_file blows up? Would the file unlink properly? Is there a way of making sure that it does?